src/main/java/dto/TagDTO.java

```diff
@@ -0,0 +1,44 @@
+package dto;
+
+import java.util.Objects;
+
+public class TagDTO {
+    private int id;
+    private String name;
+
+    public int getId() {
+        return id;
+    }
+
+    public void setId(int id) {
+        this.id = id;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    public static TagDTO createTag(int id, String name) {
+        TagDTO tagDTO = new TagDTO();
+        tagDTO.setId(id);
+        tagDTO.setName(name);
+        return tagDTO;
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        TagDTO tagDTO = (TagDTO) o;
+        return id == tagDTO.id && Objects.equals(name, tagDTO.name);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(id, name);
+    }
```

Comment:
Where do you use hashcode and equals?

Reply:
deleted the hashcode.
Equals remained because we are not comparing links but objects